### PR TITLE
feat: add language detection

### PR DESCRIPTION
**Pull request by melikhov-dev**
This PR adds the ability to detect the user's language.

Language definition scheme

1.  The language is determined by the tld. The mapping is set in the `appLangByTld` configuration field
2. This behavior is overridden through the function of direct mapping of the domain name to the language — `appGetLangByHostname` 
3. If there is an `accept-language` header, the language specified in it takes precedence over the mapping to the domain name.

It is also possible to set the language via a query parameter defined in the `appGetLangByHostname` configuration field.

When setting the language, it is checked that it is in the list of valid ones — `appAllowedLangs`.

The result is saved in the Request Context:

```ts
import {USER_LANGUAGE_PARAM_NAME} from '@gravity-ui/nodekit';

ctx.get(USER_LANGUAGE_PARAM_NAME);
```

Example

```ts
  const nodekit = new NodeKit({
        config: {
            appDefaultLang: 'ru',
            appAllowedLangs: ['ru', 'en'],
            appLangByTld: {com: 'en', ru: 'ru'},
             appLangQueryParamName: '_lang',
        },
    });
    const routes = {
        ....
    };

    const app = new ExpressKit(nodekit, routes);
```

### CODE REVIEW

File: src/lang/lang-middleware.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+export function setupLangMiddleware(appCtx: AppContext, expressApp: Express) {
+    const config = appCtx.config;
+    const languageConfig = config.language;
+    if (!languageConfig) {
+        return;
+    }
+
+    const {defaultLang, allowLanguages, langQueryParamName} = languageConfig;
+    if (allowLanguages && allowLanguages.length > 0 && defaultLang) {
```
Comment (goshander) [inline src/lang/lang-middleware.ts:14]: Maybe we can use return if one of this is null

File: src/lang/lang-middleware.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    {loose: true},
+                );
+
+                if (langByHeader) {
+                    setLang({lang: langByHeader, ctx: req.ctx});
+                }
+            }
+
+            const langQuery = langQueryParamName && req.query[langQueryParamName];
+            if (langQuery && typeof langQuery === 'string' && allowLanguages.includes(langQuery)) {
```
Comment (goshander) [inline src/lang/lang-middleware.ts:47]: If langQueryParam override lang with highest priority we can check it first and do `return next()` immediately

File: src/lang/lang-middleware.ts (hunk 1 above)
Comment (melikhov-dev) [reply src/lang/lang-middleware.ts]: yep, fixed it

File: src/lang/lang-middleware.ts (hunk 2 above)
Comment (melikhov-dev) [reply src/lang/lang-middleware.ts]: Great idea, done

**Review by goshander — APPROVED** [review]

**Review by resure — APPROVED** [review]

**Review by resure — CHANGES_REQUESTED** [review]
- app* prefix for new settings
- pr description

**Comment by melikhov-dev** [comment]
@resure done

**Review by resure — APPROVED** [review]

**Review by DakEnviy — APPROVED** [review]

**Pull request merged**